presto-parquet/src/main/java/io/prestosql/parquet/reader/TimestampColumnReader.java

```diff
@@ -40,13 +47,22 @@ public TimestampColumnReader(RichColumnDescriptor descriptor, DateTimeZone timeZ
     protected void readValue(BlockBuilder blockBuilder, Type type)
     {
         if (definitionLevel == columnDescriptor.getMaxDefinitionLevel()) {
-            long utcMillis = getTimestampMillis(valuesReader.readBytes());
             if (type instanceof TimestampWithTimeZoneType) {
-                type.writeLong(blockBuilder, packDateTimeWithZone(utcMillis, UTC_KEY));
+                int precision = ((TimestampWithTimeZoneType) type).getPrecision();
+                DecodedTimestamp decodedTimestamp = decode(valuesReader.readBytes());
+                if (precision <= TimestampWithTimeZoneType.MAX_SHORT_PRECISION) {
+                    long utcMillis = decodedTimestamp.getEpochSeconds() * MILLISECONDS_PER_SECOND + decodedTimestamp.getNanosOfSecond() / NANOSECONDS_PER_MILLISECOND;
+                    type.writeLong(blockBuilder, packDateTimeWithZone(utcMillis, UTC_KEY));
```

Comment:
We need to round/trunc the value if the file contains higher precision than the one expected by us.

please add new tests for that, as this seems have escaped

same for the long case below